olm-catalog/serverless-operator/generate_csv.sh

```diff
@@ -0,0 +1,72 @@
+# !/bin/bash
+
+template=$1
+target=$2
+
+declare -A images
+images=(
+  ["queue-proxy"]="registry.svc.ci.openshift.org/openshift/knative-v0.16.0:knative-serving-queue"
+  ["activator"]="registry.svc.ci.openshift.org/openshift/knative-v0.16.0:knative-serving-activator"
+  ["autoscaler"]="registry.svc.ci.openshift.org/openshift/knative-v0.16.0:knative-serving-autoscaler"
+  ["autoscaler-hpa"]="registry.svc.ci.openshift.org/openshift/knative-v0.16.0:knative-serving-autoscaler-hpa"
+  ["controller"]="registry.svc.ci.openshift.org/openshift/knative-v0.16.0:knative-serving-controller"
+  ["webhook"]="registry.svc.ci.openshift.org/openshift/knative-v0.16.0:knative-serving-webhook"
+  ["storage-version-migration-serving-0.16.0__migrate"]="registry.svc.ci.openshift.org/openshift/knative-v0.16.0:knative-serving-storage-version-migration"
+
+  ["3scale-kourier-gateway"]="docker.io/maistra/proxyv2-ubi8:1.1.0"
+  ["3scale-kourier-control"]="registry.svc.ci.openshift.org/openshift/knative-v0.16.0:kourier"
+
+  ["eventing-controller__eventing-controller"]="registry.svc.ci.openshift.org/openshift/knative-v0.16.0:knative-eventing-controller"
+  ["sugar-controller__controller"]="registry.svc.ci.openshift.org/openshift/knative-v0.16.0:knative-eventing-sugar-controller"
+  ["eventing-webhook__eventing-webhook"]="registry.svc.ci.openshift.org/openshift/knative-v0.16.0:knative-eventing-webhook"
+  ["storage-version-migration-eventing__migrate"]="registry.svc.ci.openshift.org/openshift/knative-v0.16.0:knative-eventing-storage-version-migration"
+
+  # the next three images have 0 replicas, they got removed, and point to 0.15 release images to migrate their deployments to 0
+  ["broker-controller__broker-controller"]="registry.svc.ci.openshift.org/openshift/knative-v0.15.2:knative-eventing-channel-broker"
+  ["broker-filter__filter"]="registry.svc.ci.openshift.org/openshift/knative-v0.15.2:knative-eventing-broker-filter"
+  ["broker-ingress__ingress"]="registry.svc.ci.openshift.org/openshift/knative-v0.15.2:knative-eventing-broker-ingress"
+
+  # the mt broker replaces the removed broker
+  ["mt-broker-controller__mt-broker-controller"]="registry.svc.ci.openshift.org/openshift/knative-v0.16.0:knative-eventing-mtchannel-broker"
```

Comment:
wouldn't it make sense to parameterize the version here (v0.16.0) as well ? (i.e. putting it into a variable)

I would do that for any duplicated parts (like the registry, too and the organisation)

Reply:
It makes sense for some parts indeed, yes. The version specifically though isn't necessarily applicable, not even the registry is. Note how we have differing patch versions between eventing and serving sometimes and how we even need a docker.io image

Reply:
Got an idea though, trying.

Reply:
Have a look again!